src/state/ducks/todos/todoSlice.ts

```diff
@@ -0,0 +1,125 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { ITodosState } from "state/ducks/todos/types/redux";
+import { ITodo } from "state/ducks/todos/types/utils";
+const initialState: ITodosState = {
+  data: [],
+  loading: false,
+  error: null,
+};
+
+const todoSlice = createSlice({
+  name: "todos",
+  initialState,
+  reducers: {
+    getTodosRequest: (state) => {
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    },
+    getTodosSuccess: (state, action: PayloadAction<ITodo[]>) => {
+      return {
+        data: action.payload,
+        loading: false,
+        error: null,
+      };
+    },
+    getTodosFail: (state) => {
+      return {
+        ...state,
+        loading: false,
+        error: "Failed to fetch todos",
+      };
+    },
+    addTodoRequest: (state, action: PayloadAction<ITodo>) => {
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    },
+    addTodoSuccess: (state, action: PayloadAction<ITodo>) => {
+      return {
+        data: [...state.data, action.payload],
+        loading: false,
+        error: null,
+      };
+    },
+    addTodoFail: (state) => {
+      return {
+        ...state,
+        loading: false,
+        error: "Todo not added",
+      };
+    },
+    removeTodoRequest: (state, action: PayloadAction<ITodo>) => {
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    },
+    removeTodoSuccess: (state, action: PayloadAction<ITodo>) => {
+      return {
+        data: state.data.filter((e) => e._id !== action.payload._id),
+        loading: false,
+        error: null,
+      };
+    },
+    removeTodoFail: (state) => {
+      return {
+        ...state,
+        loading: false,
+        error: "Todo not removed",
+      };
+    },
+    updateTodoRequest: (state, action: PayloadAction<ITodo>) => {
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    },
+    updateTodoSuccess: (state, action: PayloadAction<ITodo>) => {
+      return {
+        data: state.data.map((e) => {
+          if (e._id === action.payload._id) {
+            return {
+              ...e,
+              title: action.payload.title,
+              completed: action.payload.completed,
+            };
+          }
+          return e;
+        }),
+        loading: false,
+        error: null,
+      };
+    },
+    updateTodoFail: (state) => {
+      return {
+        ...state,
+        loading: false,
+        error: "Todo not updated",
+      };
+    },
+  },
+});
+
+export const {
+  getTodosRequest,
+  getTodosSuccess,
+  getTodosFail,
+  addTodoRequest,
+  addTodoSuccess,
+  addTodoFail,
+  removeTodoRequest,
+  removeTodoSuccess,
+  removeTodoFail,
+  updateTodoRequest,
+  updateTodoSuccess,
+  updateTodoFail,
+} = todoSlice.actions;
```

Comment:
You are creating an error state but there is no way to clear it. Try creating a fake error and see how to clear it. Also, use an alert library to show alerts.

Reply:
Kindly elaborate me how should i implement clear error/alert 
Now i am creating alert so how should i clear alert state 
Should i create one more action clearAlertState which is called on alert close  in which we reset the alert state ?